public/app/features/dashboard/api/publicDashboardApi.ts

```diff
@@ -69,40 +70,59 @@ export const publicDashboardApi = createApi({
     }),
     createPublicDashboard: builder.mutation<
       PublicDashboard,
-      { dashboard: DashboardModel; payload: Partial<PublicDashboardSettings> }
+      { dashboard: DashboardModel | DashboardScene; payload: Partial<PublicDashboardSettings> }
     >({
       query: (params) => ({
-        url: `/dashboards/uid/${params.dashboard.uid}/public-dashboards`,
+        url: `/dashboards/uid/${
+          params.dashboard instanceof DashboardScene ? params.dashboard.state.uid : params.dashboard.uid
+        }/public-dashboards`,
         method: 'POST',
         data: params.payload,
       }),
       async onQueryStarted({ dashboard, payload }, { dispatch, queryFulfilled }) {
         const { data } = await queryFulfilled;
         dispatch(notifyApp(createSuccessNotification('Dashboard is public!')));
 
-        // Update runtime meta flag
-        dashboard.updateMeta({
-          publicDashboardUid: data.uid,
-          publicDashboardEnabled: data.isEnabled,
-        });
+        if (dashboard instanceof DashboardScene) {
+          dashboard.setState({
+            meta: { ...dashboard.state.meta, publicDashboardEnabled: data.isEnabled, publicDashboardUid: data.uid },
```

Comment:
I think the `Public` tag in the navbar is not yet implemented, right? 
That's how the dashboard knows the public dashboard was created and shows that tag

Reply:
@juanicabanas yes, that's true

Reply:
should it be implemented in this PR?

Reply:
I think I'll leave it for a separate PR